### PR TITLE
feat(debug): add source maps to evaluates in debug mode

### PR DESCRIPTION
**Pull request by dgozman**
When `PLAYWRIGHT_DEBUG_UI` is set, we try to find the source of the function in the current file and source map it.

### CODE REVIEW

File: src/javascript.ts [hunk 1]
```diff
@@ -242,5 +253,101 @@ export async function prepareFunctionCall<T>(
...
+}
+
+function generateSourceMapUrl(functionText: string, compiledPosition: sourceMap.Position): string {
+  if (!getFromENV('PLAYWRIGHT_DEBUG_UI'))
+    return generateSourceUrl();
+  const filePath = getCallerFilePath();
+  if (!filePath)
+    return generateSourceUrl();
+  try {
+    const source = fs.readFileSync(filePath, 'utf8');
```
Comment (aslushnikov) [inline src/javascript.ts:284]: let's read async

**Review by aslushnikov — APPROVED** [review]
awesome! The only mild concern I have is that in `PLAYWRIGHT_UI_DEBUG` mode, we'll have an overhead of file system read and newline parsing on every evaluate call.

It might be negligible though since debugging is usually concerned with a single test.

File: src/javascript.ts (hunk 1 above)
Comment (dgozman) [reply src/javascript.ts:284]: Done.

File: src/helper.ts [hunk 2]
```diff
@@ -370,6 +370,31 @@ export function logPolitely(toBeLogged: string) {
     console.log(toBeLogged);  // eslint-disable-line no-console
 }
 
+export function getCallerFilePath(ignorePrefix = __dirname): string | null {
```
[the remark this answers is not in the document]
Comment (dgozman) [reply src/helper.ts:373]: Done.

**Comment by pavelfeldman** [comment]
@dgozman, @aslushnikov : could you elaborate the user scenario? It adds quite some complexity to the code and I can't figure out what the 'current file' means.

**Pull request merged**